htlcswitch/quiescer.go

```diff
@@ -306,3 +338,18 @@ func (q *quiescer) resume() {
 	q.received = false
 	q.resumeQueue = nil
 }
+
+type quiescerNoop struct{}
```

Comment:
🤔I think it means `quiescer` should be an optional param on `link` and we optionally init it when the feature manager says so?

Reply:
I like that less because it means that we have to do option checking everywhere and this method prevents a bunch of branching in the main line of logic.

Reply:
I prefer the option with the least branching in mainline code. We should have done this with STC's, just look at how much edge case branching there is for MuSig2 stuff.

Reply:
>  do option checking everywhere

You'd just map over it everytime. We do the same for the optional interfaces for the custom chan stuff.

Reply:
It's not quite that simple as we may be querying its state and making decisions in the link based off of that state. Instead of mapping you will almost certainly need to call ElimOption in a lot of places which is the essence of what I'm saying when I say "option checking".

We can go this route, it just kind of makes things a lot more branch-y. It seems that the team has preferred "linear code" and this makes the link code much more linear. Branching can make the code much less readable especially if one of those branches will rarely ever be called (the noop case here)